app/javascript/components/explore/FacetComponents.jsx

```diff
@@ -0,0 +1,289 @@
+      data-analytics-name="reset-cell-filters"
+      data-toggle="tooltip"
+      data-original-title="Reset filter selections"
+    >
+      <FontAwesomeIcon icon={faUndo}/>
+    </a>
+  )
+}
+
+/** Toggle icon for collapsing a list; for each filter list, and all filter lists */
+function CollapseToggleChevron({ isCollapsed, whatToToggle }) {
+  let toggleIcon
+  let toggleIconTooltipText
+  if (!isCollapsed) {
+    toggleIcon = <FontAwesomeIcon className="chevron-down" icon={faChevronDown} />
+    toggleIconTooltipText = `Hide ${whatToToggle}`
+  } else {
+    toggleIcon = <FontAwesomeIcon className="chevron-right" icon={faChevronRight} />
+    toggleIconTooltipText = `Show ${whatToToggle}`
+  }
+
+  return (
+    <span
+      className="facet-toggle-chevron"
+      data-original-title={toggleIconTooltipText}
+      {...tooltipAttrs}
+    >
+      {toggleIcon}
+    </span>
+  )
+}
+
+/** UI control to update how filters are sorted */
+function SortFiltersIcon({ sortKey, setSortKey }) {
+  const icon = sortKey === 'count' ? faSortAmountDown : faSortAlphaDown
+  const nextSortKey = sortKey === 'count' ? 'label' : 'count'
+
+  return (
+    <span
+      onClick={() => {
+        setSortKey(nextSortKey)
+        log('sort-filters', { sortKey })
+      }}
+      className={`sort-filters sort-filters-${sortKey}`}
+      data-analytics-name={`sort-filters sort-filters-${sortKey}`}
+      {...tooltipAttrs}
+      data-original-title={`Sorted by ${sortKey}; click to sort by ${nextSortKey}`}
+    >
+      <FontAwesomeIcon icon={icon}/>
+    </span>
+  )
+}
+
+/**
+ * Very brief notes on terms in SCP metadata convention, derived from:
+ * https://singlecell.zendesk.com/hc/en-us/articles/360060609852-Required-Metadata
+ */
+const conventionalMetadataGlossary = {
+  'biosample_id': 'Unique identifier for each sample in the study',
+  'donor_id': 'Unique identifier for each biosample donor in the study',
+  'species__ontology_label': 'Taxon name, from NCBITaxon',
+  'disease__ontology_label': 'Disease name, from Mondo or PATO',
+  'organ__ontology_label': 'Organ name, from Uberon',
+  'library_preparation_protocol__ontology_label': 'From EFO',
+  'sex': 'One of "female", "male", "mixed", or "unknown"',
+  'cell_type__ontology_label': 'From Cell Ontology',
+  'ethnicity__ontology_label': 'From Human Ancestry Ontology'
+}
+
+/** Determine if annotation is conventional from its raw name */
+function getIsConventionalAnnotation(rawName) {
+  return (
+    rawName.includes('__ontology_label') ||
+    ['donor_id', 'biosample_id'].includes(rawName)
+  )
+}
+
+/** Determine if a list of DOM classes includes one used for the sort-icon */
+function includesSortIconClass(domClasses) {
+  return (
+    domClasses.includes('fa-sort-alpha-down') ||
+    domClasses.includes('fa-sort-amount-down') ||
+    domClasses.includes('sort-filters')
+  )
+}
+
+/** Convert e.g. "cell_type__ontology_label" to "Cell type" */
+function parseAnnotationName(annotationIdentifier) {
```

Comment:
Question: would it be possible to apply a function like this to the options dropdown for annotations? It's a lovely refinement that makes our required metadata names much more readable. It'd be great if portal UI could use the same name transformation for our clunky *__ontology_label metadata names throughout the Explore tab (options panel, legend title, cell filtering etc)

Reply:
Makes sense!  SCP-5538 now tracks that.